core/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -345,6 +345,9 @@ public boolean getShouldStoreResult() {
      * The number of slices this task should be divided into. Defaults to 1 meaning the task isn't sliced into subtasks.
      */
     public Self setSlices(int slices) {
+        if (slices < 0) {
+            throw new IllegalArgumentException("[slices] must be at least 0");
```

Comment:
Just in case this is ever thrown it'd be nice to have the error message more like "slices must be at least 0 but was [$value]".